packages/pages/src/vite-plugin/modules/plugin.ts

```diff
@@ -61,6 +62,14 @@ export const buildModules = async (
     );
   }
 
+  const viteConfig = await import(
+    scopedViteConfigPath(projectStructure.config.scope) ?? ""
+  );
+  const config = viteConfig?.default;
+  const filteredPlugins = config?.plugins?.[0]?.filter(
```

Comment:
Don't you only want to remove the pages plugin? I think we can use mergeConfig instead and you set the below build config to an object and pass it in as the overrides - https://vitejs.dev/guide/api-javascript#mergeconfig

Reply:
yay used that. also prevented the looping issue